code/modules/mob/living/carbon/human/life.dm

```diff
@@ -64,6 +64,10 @@
 
 	if(chest_covered && head_covered)
 		return ONE_ATMOSPHERE
+	if(istype(loc,/obj/structure/closet))
+		/// If we're in a closet with 0.5 pressure protection, it averages the values, for example.
+		var/obj/structure/closet/current_locker = loc
+		return (current_locker.pressure_protection * ONE_ATMOSPHERE + (1 - current_locker.pressure_protection) * pressure)
```

Comment:
You should not have this code in the human type, you could probably best do this with a signal and having the closets register for it.

Reply:
so looking at this closer it looks like you should be making a pressure_protection and thermal_insulation var on... /atom/movable? And in the proc for mobs you should be doing something like

```
var/atom/movable/target = mob.loc
var/total_insulation = 0
while(ismovable(target))
  total_insulation += (1 - total_insulation) * target.insulation
  target = target.loc
```